File: test/language/subtypes/cpp/SubtypeImportedTest.cpp
#include "gtest/gtest.h"

#include "subtypes/subtype_imported/SubtypeImported.h"

namespace subtypes
{
namespace subtype_imported
{

TEST(SubtypeImportedTest, readWrite)
{
    SubtypeImported subtypeImported{pkg::SubTest{42}};

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    subtypeImported.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    SubtypeImported readSubtypeImported(reader);

    ASSERT_TRUE(subtypeImported == readSubtypeImported);
}

} // namespace subtype_imported
} // namespace subtypes

File: test/language/subtypes/python/SubtypeImportedTest.py
import zserio

import Subtypes


class SubtypeImportedTest(Subtypes.TestCase):
    def testReadWrite(self):
        subtypeImported = self.api.SubtypeImported(self.api.pkg.SubTest(13))

        writer = zserio.BitStreamWriter()
        subtypeImported.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readSubtypeImported = self.api.SubtypeImported()
        readSubtypeImported.read(reader)
        self.assertEqual(subtypeImported, readSubtypeImported)
